### ISSUE TITLE
New option allow-multiline on comma-dangle

### ISSUE THREAD

**Issue by gimenete** [eslint/eslint#4967]
I like using the dangle comma on multiline arrays and objects but I don't like the linter warn me if I don't use it or any of my coworkers don't use it.

In other words I would love an option to make the dangle comma optional on multiline nodes. So this is what this PR is about.

This would allow also to adopt this option on standard as mentioned here feross/standard#240

I have code I'm going to push in a PR

**Comment by eslintbot** [comment]
@gimenete Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
Can you show some examples and provide an over view of what the change to the rule would be?

**Comment by gimenete** [comment]
Sure. Basically this option accepts these patterns:

```javascript
var foo = {
    bar: "baz",
    qux: "quux",
};

var foo = {
    bar: "baz",
    qux: "quux"
};
```

And these are considered problems:

```javascript
var foo = { bar: "baz", qux: "quux", }; /*error Unexpected trailing comma.*/
var arr = [1,2,];
```

So the dangle comma is accepted for multiline nodes but it is not mandatory.
